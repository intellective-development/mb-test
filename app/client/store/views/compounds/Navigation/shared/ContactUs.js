// @flow

import * as React from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';
import I18n from 'store/localization';
import { MBDynamicIcon, MBGrid, MBLayout, MBLink, MBModal, MBText } from 'store/views/elements';
import { ui_actions, ui_selectors } from 'store/business/ui';
import { working_hours_selectors } from '../../../../business/working_hours';
import styles from './ContactUs.scss';
import type { WorkingHour } from '../../../../business/working_hours';
import gropedWorkingHours from '../../../../../utils/group_working_hours';

type ContactUsProps = {
  hideModal: typeof ui_actions.hideHelpModal,
  show: boolean,
  working_hours: WorkingHour
};

const ContactUs = ({ hideModal, show, working_hours }: ContactUsProps) => {
  const icon_size = 48;

  return (
    <MBModal.Modal
      show={show}
      onHide={hideModal}
      size="small"
      modalClassName={styles.cmContactUs_Modal}>
      <MBModal.SectionHeader
        renderRight={() => <MBModal.Close onClick={hideModal} />}>
        {I18n.t('ui.nav.help.header')}
      </MBModal.SectionHeader>
      <MBLayout.StandardGrid>
        <div className={styles.cmContactUs_Wrapper}>
          <MBGrid cols={1} medium_cols={2}>
            <MBGrid.Element className={styles.cmContactUs_Section_Wrapper}>
              <span className={styles.cmContactUs_Section}>
                <MBDynamicIcon className={styles.cmContactUs_Section_Icon} name="faq" color="black" height={icon_size} width={icon_size} />
                <MBText.H6 className={styles.cmContactUs_Section_Header}>{I18n.t('ui.nav.help.faqs.title')}</MBText.H6>
                <MBLink.Text target="_blank" rel="noopener noreferrer" native_behavior href="https://minibar.freshdesk.com/support/home" className={styles.cmContactUs_Section_Link}>{I18n.t('ui.nav.help.faqs.link')}</MBLink.Text>
                <MBText.Span className={styles.cmContactUs_Section_Title}>{I18n.t('ui.nav.help.faqs.info')}</MBText.Span>
              </span>
            </MBGrid.Element>
            <MBGrid.Element className={styles.cmContactUs_Section_Wrapper}>
              <span className={styles.cmContactUs_Section}>
                <MBDynamicIcon className={styles.cmContactUs_Section_Icon} name="representative" color="black" height={icon_size} width={icon_size} />
                <MBText.H6 className={styles.cmContactUs_Section_Header}>{I18n.t('ui.nav.help.customer_care.title')}</MBText.H6>
                {working_hours &&
                  gropedWorkingHours(working_hours).map(wh => {
                    return (
                      <span key={wh.startDay + wh.endDay}>
                        <MBText.Span className={styles.cmContactUs_Section_Title}>{I18n.t('ui.nav.help.customer_care.hours', { startsAt: _.toUpper(wh.startsAt), endsAt: wh.endsAt === '11:59 pm' ? 'Midnight' : _.toUpper(wh.endsAt) })}</MBText.Span><br />
                        <MBText.Span className={styles.cmContactUs_Section_Title}>{wh.startDay === wh.endDay ? I18n.t('ui.nav.help.customer_care.info', { info: wh.startDay }) : I18n.t('ui.nav.help.customer_care.info', { info: `${wh.startDay} - ${wh.endDay}` })}</MBText.Span>
                      </span>);
                  })
                }
              </span>
            </MBGrid.Element>
          </MBGrid>
          <MBText.H5 className={styles.cmContactUs_Divider}>{I18n.t('ui.nav.help.contact.headline')}</MBText.H5>
          <MBGrid cols={1} medium_cols={3}>
            <MBLink.View native_behavior href="sms:917-633-6332" className={styles.cmContactUs_Section_Button}>
              <MBGrid.Element className={styles.cmContactUs_Section_Wrapper}>
                <span className={styles.cmContactUs_Section}>
                  <MBDynamicIcon className={styles.cmContactUs_Section_Icon} name="sms" color="black" height={icon_size} width={icon_size} />
                  <MBText.H6 className={styles.cmContactUs_Section_Title}>{I18n.t('ui.nav.help.contact.text.title')}</MBText.H6>
                  <MBLink.Text>{I18n.t('ui.nav.help.contact.text.data')}</MBLink.Text>
                </span>
              </MBGrid.Element>
            </MBLink.View>
            <MBLink.View native_behavior href="mailto:help@minibardelivery.com" className={styles.cmContactUs_Section_Button}>
              <MBGrid.Element className={styles.cmContactUs_Section_Wrapper}>
                <span className={styles.cmContactUs_Section}>
                  <MBDynamicIcon className={styles.cmContactUs_Section_Icon} name="email" color="black" height={icon_size} width={icon_size} />
                  <MBText.H6 className={styles.cmContactUs_Section_Title}>{I18n.t('ui.nav.help.contact.email.title')}</MBText.H6>
                  <MBLink.Text>{I18n.t('ui.nav.help.contact.email.data')}</MBLink.Text>
                </span>
              </MBGrid.Element>
            </MBLink.View>
            <MBLink.View native_behavior href="tel:855-487-0740" className={styles.cmContactUs_Section_Button}>
              <MBGrid.Element className={styles.cmContactUs_Section_Wrapper}>
                <span className={styles.cmContactUs_Section}>
                  <MBDynamicIcon className={styles.cmContactUs_Section_Icon} name="phone" color="black" height={icon_size} width={icon_size} />
                  <MBText.H6 className={styles.cmContactUs_Section_Title}>{I18n.t('ui.nav.help.contact.phone.title')}</MBText.H6>
                  <MBLink.Text>{I18n.t('ui.nav.help.contact.phone.data')}</MBLink.Text>
                </span>
              </MBGrid.Element>
            </MBLink.View>
          </MBGrid>
        </div>
      </MBLayout.StandardGrid>
    </MBModal.Modal>
  );
};

const ContactUsSTP = (state) => ({
  show: ui_selectors.isHelpModalShowing(state),
  working_hours: working_hours_selectors.workingHours(state)
});
const ContactUsDTP = { hideModal: ui_actions.hideHelpModal };

const ContactUsContainer = connect(ContactUsSTP, ContactUsDTP)(ContactUs);

export default ContactUsContainer;
