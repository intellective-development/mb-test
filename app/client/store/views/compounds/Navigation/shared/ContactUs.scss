@import "~client/store/views/style/variables";

.cmContactUs_Modal {
  width: 736px;

  @media only screen and (max-width: $small-max-width) {
    width: 80%;
    margin-bottom: 16px;
  }
}
.cmContactUs_Wrapper {
  padding: 2em 3em;

  @media only screen and (max-width: $small-max-width) {
    padding: 2em;
  }
}

.cmContactUs_Section {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 1em;
  text-align: center;
}

.cmContactUs_Section_Header {
  @include font-size(15);

  font-weight: $heavy-font-weight;
}

.cmContactUs_Section_Icon {
  margin-bottom: 5px;
}

.cmContactUs_Section_Title {
  @include font-size(13);
}

.cmContactUs_Section_Link,
.cmContactUs_Section_Button {
  @include font-size(13);

  font-weight: $heavy-font-weight;
}

.cmContactUs_Section_Button:hover {
  background-color: $brand-pale-blue;
  border-radius: 4px;
}

.cmContactUs_Divider {
  @include font-size(14);
  @include hr-inset-text(14, 1px, $brand-medium-grey, 1em);

  letter-spacing: 2px;
  padding: 1em 0;
}
