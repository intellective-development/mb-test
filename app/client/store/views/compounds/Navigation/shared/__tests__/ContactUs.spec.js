
import React from 'react';

import TestProvider from 'store/views/__tests__/utils/TestProvider';
import ContactUs from '../ContactUs';

describe('ContactUs', () => {
  it.skip('renders', () => {
    // TODO: test in the branch the contains the actual implementation

    const initial_state = {
      ui: {
        show_help_modal: true
      }
    };

    expect(render(
      <TestProvider initial_state={initial_state}>
        <ContactUs />
      </TestProvider>
    )).toMatchSnapshot();
  });
});
