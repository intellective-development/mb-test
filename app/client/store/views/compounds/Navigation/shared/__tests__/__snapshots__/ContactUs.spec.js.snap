// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUs renders 1`] = `null`;
